mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -100,4 +152,37 @@ public DBObject parse(String json) {
 		return (DBObject) object.get( "json" );
 	}
 
+	private static DBObject stage(String key, Object value) {
+		DBObject stage = new BasicDBObject();
+		stage.put( key, value );
+		return stage;
+	}
+
+	public boolean addPipeline(StackElement element, String value) {
+		if ( !parsed.contains( element.getIndex() ) ) {
+			parsed.add( element.getIndex() );
+			boolean isJsonObject = value.trim().matches( "^\\{.*:.*\\}$" );
```

Comment:
The parse method doesn't work for some cases because it expects the return value to be a DBObject but now sometimes it returns a String. I will change the approach